model/change_request_metadata.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package model
+
+import (
+	"encoding/json"
+)
+
+// ChangeRequest is the Ring change request stored in a model.Ring.
+type ChangeRequest struct {
+	Image        string
+	Version      string
+	Hotfix       bool
+	ReleaseStart int64
+}
+
+// NewChangeRequestMetadata creates an instance of ChangeRequestMetadata given the raw elrond change request.
+func NewChangeRequestMetadata(metadataBytes []byte) (*ChangeRequest, error) {
+	// Check if length of metadata is 0 as opposed to if the value is nil. This
+	// is done to avoid an issue encountered where the metadata value provided
+	// had a length of 0, but had non-zero capacity.
+	if len(metadataBytes) == 0 || string(metadataBytes) == "null" {
+		// TODO: remove "null" check after sqlite is gone.
+		return nil, nil
```

Comment:
Yeah, I personally strongly prefer not using pointers unless explicitly representing an absence of something. Pointer performance impact might actually be negative since it is more work for GC. Unforunatelly in Provisioner we maintain this pattern for consistency with older code, which IMO is more important.

In this case if all `New.*` functions return pointers I would stick with that for the sake of consistency (unless we want to change the all, then 💯 ), as an aside I would propose getting rid of the `ChangeRequest` all together, see my other comments.